pkg/distributor/field_detection.go

```diff
@@ -123,13 +137,24 @@ func (l *LevelDetector) detectLogLevelFromLogEntry(entry logproto.Entry, structu
 	return l.extractLogLevelFromLogLine(entry.Line)
 }
 
-func (l *LevelDetector) extractLogLevelFromLogLine(log string) string {
-	logSlice := unsafe.Slice(unsafe.StringData(log), len(log))
+func (l *FieldDetector) detectGenericFieldFromLogEntry(entry logproto.Entry, hints []string) string {
+	lineBytes := unsafe.Slice(unsafe.StringData(entry.Line), len(entry.Line))
+	var v []byte
+	if isJSON(entry.Line) {
+		v = l.getValueUsingJSONParser(lineBytes, hints)
+	} else if isLogFmt(lineBytes) {
+		v = l.getValueUsingLogfmtParser(lineBytes, hints)
+	}
+	return string(v)
+}
+
+func (l *FieldDetector) extractLogLevelFromLogLine(log string) string {
```

Comment:
Couldn't we drop this function & just use `detectGenericFieldFromLogEntry`?

Reply:
No, log level detection is slightly different.